go_v2/main.go

```diff
@@ -0,0 +1,84 @@
+/*
```

Comment:
I've altered the origin repo to have a MIT license at the root, not sure if I've done it correctly. Top level plus reference to each file I think would be a bit leaner. 

You may need to pull from upstream first if you want to update anything I've changed since your fork.